Source/Charts/Renderers/LineChartRenderer.swift

```diff
@@ -881,7 +881,15 @@ open class LineChartRenderer: LineRadarRenderer
         context.addPath(spline)
         context.replacePathWithStrokedPath()
         context.clip()
-        context.drawLinearGradient(gradient, start: gradientStart, end: gradientEnd, options: [])
+        
+        switch dataSet.gradientType {
```

Comment:
Since the only thing changing is the `start` and `end`, the switch statement should be there i.e.
```
let start, end: CGPoint
switch dataSet.gradientType {
    case .horizontal: 
        start = ...
        end = ...

    case .vertical: ...
}
```